public/apps/configuration/panels/role-mapping/RoleEditMappedUser.tsx

```diff
@@ -62,11 +61,11 @@ export function RoleEditMappedUser(props: RoleEditMappedUserProps) {
   useEffect(() => {
     const fetchData = async () => {
       try {
-        const originalRoleMapData = (await getRoleMappingData(
+        const originalRoleMapData: RoleMappingDetail | undefined = await getRoleMappingData(
           props.coreStart.http,
           props.roleName
-        )) as RoleMappingDetail;
-        if (!isEmpty(originalRoleMapData)) {
+        );
+        if (originalRoleMapData) {
```

Comment:
isEmpty is a strong empty check. Just want to confirm that this is the intentional logic change.

Reply:
Yep, the original getRoleMappingData returns `[]` when 404. So it relies on `isEmpty` to check. After change it returns undefined, so just check if it is true should be fine.